hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java

```diff
@@ -220,7 +223,11 @@ public Expression resolveSqlExpression(
 
 		pushProcessingState( restrictionProcessingState, getFromClauseIndex() );
 		try {
-			return (Predicate) sqmWhereClause.getPredicate().accept( this );
+			final Predicate predicate = (Predicate) sqmWhereClause.getPredicate().accept( this );
+			if ( discriminatorPredicate != null ) {
```

Comment:
The null check shouldn't be necessary, the `combinePredicates` method handles this properly